homeassistant/components/homekit/type_cameras.py

```diff
@@ -313,51 +313,41 @@ async def start_stream(self, session_info, stream_config):
         if not opened:
             _LOGGER.error("Failed to open ffmpeg stream")
             return False
-        session_info["stream"] = stream
+
         _LOGGER.info(
             "[%s] Started stream process - PID %d",
             session_info["id"],
             stream.process.pid,
         )
 
-        ffmpeg_watcher = async_track_time_interval(
-            self.hass, self._async_ffmpeg_watch, FFMPEG_WATCH_INTERVAL
+        session_info["stream"] = stream
+        session_info[FFMPEG_PID] = stream.process.pid
+        session_info[FFMPEG_WATCHER] = async_track_time_interval(
+            self.hass,
+            lambda _: self._async_ffmpeg_watch(session_info["id"]),
```

Comment:
lambda's will be run in the executor as we can't detect that they are callbacks or async. We also can't decorate lambda's so you'll need to define the function either on the class or inline.
```suggestion
        @callback
        def watch_session(_):
            self._async_ffmpeg_watch(session_info["id"])

        session_info[FFMPEG_WATCHER] = async_track_time_interval(
            self.hass,
            watch_session,
```

Reply:
Oh, I was wrong, you can decorate lambdas.
```suggestion
        session_info[FFMPEG_WATCHER] = async_track_time_interval(
            self.hass,
            callback(lambda _: self._async_ffmpeg_watch(session_info["id"])),
```

Reply:
Actually, this is not a callback. `_async_ffmpeg_watch` is an async function and returns a coroutine. Try 3 😉 

```suggestion
        async def watch_session(_):
            await self._async_ffmpeg_watch(session_info["id"])

        session_info[FFMPEG_WATCHER] = async_track_time_interval(
            self.hass,
            watch_session,
```